tests/ethertest.nim

```diff
@@ -16,7 +16,7 @@ template ethersuite*(name, body) =
     var snapshot: JsonNode
 
     setup:
-      ethProvider = JsonRpcProvider.new("ws://localhost:8545")
+      ethProvider = JsonRpcProvider.new("http://localhost:8545")
```

Comment:
I think this is the right change, but we also might run into the `poll` future failing... (something to be fixed in ethers)

Reply:
You're right, the contract tests started failing with this change. Strange that works in the integration tests, but not in the contract tests. I'll revert the change on this line for now.

Reply:
Hard to know if the reason they are failing is due to the `poll` failing issue though. In saying that, I'd like to get a fix in for the `poll()` fail. Trying to work on it now, but unsure if I'll get something in before my flight leaves in a few hours.